core-java-modules/core-java-sockets/src/main/java/com/baeldung/threading/connection/ConnectionHandler.java

```diff
@@ -0,0 +1,36 @@
+package com.baeldung.threading.connection;
+
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.baeldung.threading.ClientConnection;
+
+public class ConnectionHandler extends Thread {
```

Comment:
For clarity let's rename:
```suggestion
public class ThreadPerConnection extends Thread {
```